scalar/equal.go

```diff
@@ -0,0 +1,40 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package scalar
+
+// EqualPStrict returns true if two given scalar pointers are strictly
+// equal, meaning that they either have the same nility or the same value.
+func EqualPStrict[s Scalar](x, y *s) bool {
```

Comment:
I don't see a reason to limit this method to type `s Scalar`? We could realistically extend this to anything of type `comparable` - see the example in the first code block here https://go.dev/ref/spec#Instantiations

Reply:
The `return *x == *y` is only applicable to scalar values. For structs, map and arrays we will have to use different methodologies.